hwilib/serializations.py

```diff
@@ -792,8 +813,10 @@ def deserialize(self, psbt):
             input.deserialize(f)
             self.inputs.append(input)
 
-            if input.non_witness_utxo and input.non_witness_utxo.rehash() and input.non_witness_utxo.sha256 != txin.prevout.sha256:
```

Comment:
Typing not only helps for code clarity, but also helps to catch issues like this (`prevout.sha256` does not exist, and `rehash()` never returns anything).